src/main/drivers/pwm_output.c

```diff
@@ -129,18 +130,73 @@ static void pwmWriteMultiShot(uint8_t index, uint16_t value)
     *motors[index].ccr = lrintf(((float)(value-1000) * MULTISHOT_20US_MULT) + MULTISHOT_5US_PW);
 }
 
-void pwmWriteMotor(uint8_t index, uint16_t value)
+#ifdef USE_DSHOT
+static void pwmWriteDigital(uint8_t index, uint16_t value)
+{
+    pwmMotorOutput_t * const motor = &motors[index];
+
+    motor->value = value;
+    motor->updateFlags |= MOTOR_UPDATE_VALUE;
+}
+
+void pwmCompleteDigitalMotorUpdate(uint8_t motorCount)
+{
+    UNUSED(motorCount);
+
+}
+
+void pwmWriteValueToDmaBuffer(uint16_t value, uint32_t *buffer, uint8_t flags)
+{
+    buffer[0]  = (value & 0x400) ? MOTOR_BIT_1 : MOTOR_BIT_0;
```

Comment:
@borisbstyle : I forgot to initialize csum to 0 .. 
```
    uint16_t packet = (value << 1) | ((flags & MOTOR_UPDATE_TELEMETRY) ? 1 : 0);   // ESC power + update bit
    // compute checksum 
    int csum = 0;
    int csum_data = packet;
    for(int i = 0; i < 3; i++) {
        csum ^=  csum_data;   // xor data by nibbles
        csum_data >>= 4;
    }
    // append checksum
    packet= (packet << 4) | csum;
    // generate pulses for whole packet                                         
    for(int i = 0; i < 16; i++) {
        buffer[i] = (packet & 0x8000) ? MOTOR_BIT_1 : MOTOR_BIT_0;        // MSB first
        packet<<= 1;
    }
```

another possibility for checksum, slightly faster- folding the register. Extending to 32bits is only 1 additional instruction:
```
int csum = packet;
csum ^= csum >> 8;
csum ^= csum >> 4;
csum &= 0xff;